blobstore/buckets.go

```diff
@@ -95,24 +101,44 @@ func (bh *BlobHandler) HandleListBuckets(c echo.Context) error {
 			for _, b := range result.Buckets {
 				mostRecentBucketList = append(mostRecentBucketList, *b.Name)
 			}
-			if !isIdenticalArray(bh.S3Controllers[i].Buckets, mostRecentBucketList) {
-
-				bh.S3Controllers[i].Buckets = mostRecentBucketList
-
+			if !isIdenticalArray(controller.Buckets, mostRecentBucketList) {
+				controller.Buckets = mostRecentBucketList
 			}
 		}
+
 		// Extract the bucket names from the response and append to allBuckets
-		for _, bucket := range bh.S3Controllers[i].Buckets {
+		for i, bucket := range controller.Buckets {
+			permissions, fullAccessTmp, err := bh.GetUserS3ReadListPermission(c, bucket)
+			if err != nil {
+				return c.JSON(http.StatusInternalServerError, fmt.Errorf("error fetching user permissions: %s", err.Error()))
+			}
+			fullAccess = fullAccess || fullAccessTmp // Update full access based on any bucket returning full access
+
+			canRead := len(permissions) > 0 || fullAccessTmp // Set canRead based on permissions or full access
 			allBuckets = append(allBuckets, BucketInfo{
-				ID:   currentID,
-				Name: bucket,
+				ID:      i,
+				Name:    bucket,
+				CanRead: canRead,
 			})
-			currentID++ // Increment the ID for the next bucket
+		}
+	}
 
+	if fullAccess { // If full access is true, set CanRead to true for all buckets
+		for i := range allBuckets {
+			allBuckets[i].CanRead = true
 		}
 	}
-	bh.Mu.Unlock()
+
+	// Sorting allBuckets slice by CanRead true first and then by Name field alphabetically
+	sort.Slice(allBuckets, func(i, j int) bool {
```

Comment:
If I'm understanding the question correctly, then yes. It sorts buckets that start with a number first and then everything else alphabetically, since bucket.name is a string. 

![image](https://github.com/Dewberry/s3api/assets/54958829/f8112920-1bc6-40ff-afda-6b193daa4713)